// Copyright 2017 Intel Corporation
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// -----------------------------------------------------------------------------


syntax = "proto3";


message ProposeAddTaskOwner {

    string proposal_id = 1;

    string task_id = 2;

    string user_id = 3;

    string reason = 4;

    string metadata = 5;

}


message ProposeRemoveTaskOwner {

    string proposal_id = 1;

    string task_id = 2;

    string user_id = 3;

    string reason = 4;

    string metadata = 5;

}


message ProposeAddTaskAdmin {

    string proposal_id = 1;

    string task_id = 2;

    string user_id = 3;

    string reason = 4;

    string metadata = 5;

}


message ProposeRemoveTaskAdmin {

    string proposal_id = 1;

    string task_id = 2;

    string user_id = 3;

    string reason = 4;

    string metadata = 5;

}


message ConfirmAddTaskOwner {

    string proposal_id = 1;

    string task_id = 2;

    string user_id = 3;

    string reason = 4;

}


message ConfirmRemoveTaskOwner {

    string proposal_id = 1;

    string task_id = 2;

    string user_id = 3;

    string reason = 4;

}


message ConfirmAddTaskAdmin {

    string proposal_id = 1;

    string task_id = 2;

    string user_id = 3;

    string reason = 4;

}


message ConfirmRemoveTaskAdmin {

    string proposal_id = 1;

    string task_id = 2;

    string user_id = 3;

    string reason = 4;

}


message RejectAddTaskOwner {

    string proposal_id = 1;

    string task_id = 2;

    string user_id = 3;

    string reason = 4;

}


message RejectRemoveTaskOwner {

    string proposal_id = 1;

    string task_id = 2;

    string user_id = 3;

    string reason = 4;

}


message RejectAddTaskAdmin {

    string proposal_id = 1;

    string task_id = 2;

    string user_id = 3;

    string reason = 4;

}


message RejectRemoveTaskAdmin {

    string proposal_id = 1;

    string task_id = 2;

    string user_id = 3;

    string reason = 4;

}


message CreateTask {

    string task_id = 1;

    string name = 2;

    repeated string admins = 3;

    repeated string owners = 4;

    string metadata = 5;

}


message UpdateTask {

    string task_id = 1;

    string new_name = 2;

    string old_metadata_sha512 = 3;

    string new_metadata = 4;

}
